src/etl/sars-cov-2/steps/generate-consolidated-country-data-step.ts

```diff
@@ -0,0 +1,102 @@
+  StructuredPositiveCaseDataAfterJitteringPinLatLngStep;
+export type StructuredCountryPopulationDataAfterGeneratingConsolidatedCountryDataStep =
+  StructuredCountryPopulationDataAfterJitteringPinLatLngStep;
+export interface ConsolidatedCountryDataAfterGeneratingConsolidatedCountryDataStep {
+  year: number;
+  month: number;
+  alphaTwoCode: string;
+  alphaThreeCode: string;
+};
+
+
+interface GenerateConsolidatedCountryDataStepInput {
+  allEstimates: EstimateFieldsAfterJitteringPinLatLngStep[];
+  allStudies: StudyFieldsAfterJitteringPinLatLngStep[];
+  vaccinationData: StructuredVaccinationDataAfterJitteringPinLatLngStep;
+  positiveCaseData: StructuredPositiveCaseDataAfterJitteringPinLatLngStep;
+  countryPopulationData: StructuredCountryPopulationDataAfterJitteringPinLatLngStep;
+  mongoClient: MongoClient;
+}
+
+interface GenerateConsolidatedCountryDataStepOutput {
+  allEstimates: EstimateFieldsAfterGeneratingConsolidatedCountryDataStep[];
+  allStudies: StudyFieldsAfterGeneratingConsolidatedCountryDataStep[];
+  vaccinationData: StructuredVaccinationDataAfterGeneratingConsolidatedCountryDataStep;
+  positiveCaseData: StructuredPositiveCaseDataAfterGeneratingConsolidatedCountryDataStep;
+  countryPopulationData: StructuredCountryPopulationDataAfterGeneratingConsolidatedCountryDataStep;
+  consolidatedCountryData: ConsolidatedCountryDataAfterGeneratingConsolidatedCountryDataStep[];
+  mongoClient: MongoClient;
+}
+
+export const generateConsolidatedCountryDataStep = (
+  input: GenerateConsolidatedCountryDataStepInput
+): GenerateConsolidatedCountryDataStepOutput => {
+  console.log(`Running step: generateConsolidatedCountryDataStep. Remaining estimates: ${input.allEstimates.length}.`);
+
+  //Only save country data between the earliest start date and the latest end date.
+  const earliestSamplingStartDate = new Date(Math.min(...
+    input.allEstimates
+      .filter((estimate): estimate is Omit<typeof estimate, 'samplingStartDate'> & {
+        samplingStartDate: NonNullable<typeof estimate['samplingStartDate']>
+      } => !!estimate.samplingStartDate)
+      .map(({ samplingStartDate }) => samplingStartDate.valueOf())
+  ))
+  const latestSamplingEndDate = new Date(Math.max(...
+    input.allEstimates
+      .filter((estimate): estimate is Omit<typeof estimate, 'samplingEndDate'> & {
+        samplingEndDate: NonNullable<typeof estimate['samplingEndDate']>
+      } => !!estimate.samplingEndDate)
+      .map(({ samplingEndDate }) => samplingEndDate.valueOf())
+  ))
+  const earliestMonthCount = dateToMonthCount(earliestSamplingStartDate);
+  const latestMonthCount = dateToMonthCount(latestSamplingEndDate);
+  //In the second part of this issue, make this contain every single country by reading data from Airtable.
+  const allCountries = [{
+    alphaTwoCode: 'CA',
+    alphaThreeCode: 'CAN'
+  }, {
+    alphaTwoCode: 'US',
+    alphaThreeCode: 'USA'
+  }]
```

Comment:
To add a bit to the comment, the best way to get every country is to go to Airtable and read the "Countries" table. It's handy because I wanted to convert the SC2 ETL to use that table anyways. Just for this PR for demonstration purposes we're only including Canada and USA. This PR brings every country into the mix https://github.com/serotracker/iit-backend-v2/pull/29